### PR TITLE
xcode: update to 14.0

### LINKED ISSUE #2544: [CI] Update to Xcode 14
This is now the latest stable Xcode release and we should update to it in order to compile against the latest iOS SDK.

* Version: 14.0 (14A309)
* Release Notes: https://developer.apple.com/documentation/xcode-release-notes/xcode-14-release-notes
* Download: https://download.developer.apple.com/Developer_Tools/Xcode_14/Xcode_14.xip

Will require EngFlow to update their executors just as was done for #1997 and #2100. This version of Xcode also requires macOS 12.5 to run, so an OS update to the workers may also be needed.

cc @lfpino @Yannic

### PR DESCRIPTION
**Pull request by jpsim**
Updates the version of Xcode Envoy Mobile is compatible with to the latest stable version.

Risk Level: Low.
Testing: Local & CI.
Docs Changes: Done.
Release Notes: Done.
Fixes #2544

Signed-off-by: JP Simard <jp@jpsim.com>

### CODE REVIEW

File: ci/platform/BUILD [hunk 1]
```diff
@@ -7,6 +7,6 @@ platform(
         "@platforms//os:macos",
     ],
     exec_properties = {
-        "Pool": "macos",
+        "Pool": "xcode_14",
```
Comment (jpsim) [inline ci/platform/BUILD:10]: This is temporary and should be reverted before merging. /cc @Yannic

**Comment by jpsim** [comment]
Interesting FlatBuffers compiler bug (I think):

```
bazel-out/ios-x86_64-min12.0-applebin_ios-ios_x86_64-fastbuild-ST-5ef840230b88/bin/external/swift_flatbuffers/swift/FlatBuffers.swiftinterface:161:3: error: missing return in accessor expected to return 'Swift.Int'  }  ^bazel-out/ios-x86_64-min12.0-applebin_ios-ios_x86_64-fastbuild-ST-5ef840230b88/bin/external/swift_flatbuffers/swift/FlatBuffers.swiftinterface:1:1: error: failed to verify module interface of 'FlatBuffers' due to the errors above; the textual interface may be broken by project issues or a compiler bug// swift-interface-format-version: 1.0^
```

**Comment by jpsim** [comment]
I've filed https://github.com/apple/swift/issues/61455 to report the corrupt `.swiftinterface` file.

**Comment by jpsim** [comment]
Proposing dropping FlatBuffers: https://github.com/envoyproxy/envoy-mobile/pull/2594

**Comment by jpsim** [comment]
@Augustyniak this is now passing CI and ready for review.

**Review by Augustyniak — DISMISSED** [review]

**Comment by jpsim** [comment]
@Augustyniak sorry once more now that I've updated with `main`.

**Review by Augustyniak — APPROVED** [review]

**Pull request merged**